plone/supermodel/exportimport.py

```diff
@@ -397,7 +397,9 @@ def write(self, field, name, type, elementName='field'):
         ):
             value = []
             for term in field.vocabulary:
-                if (
+                if isinstance(term.value, six.integer_types):
+                    pass
+                elif (
                     not isinstance(term.value, six.string_types) or
                     six.b(term.token) != term.value.encode('unicode_escape')
                 ):
```

Comment:
Wow, the whole method is horrible to read. Let me see....

Reply:
The condition is a bit complicated to read. I tried to include the integer-condition in the existing condition, but I couldn't do it in a readable way.
The point of the condition is that the encode should only be executed when it is a string type.
I found that my suggested change is quite readable with minimum change in code.
I'm open for better / more readable solutions 😉

Reply:
What about the positive expression of it?

``` python
            for term in field.vocabulary:
                if (
                    isinstance(term.value, six.integer_types)
                    or (
                        isinstance(term.value, six.string_types)
                        and six.b(term.token) == term.value.encode('unicode_escape')
                    )
                ):
                    if term.title and term.title != term.value:
                        value.append((term.value, term.title))
                    else:
                        value.append(term.value)
                else:
                    raise NotImplementedError(
                        u"Cannot export a vocabulary that is not "
                        u"based on a simple list of values"
                    )
```

Reply:
I like that 👍 I have updated the PR.